dashboard/app/controllers/levels_controller.rb

```diff
@@ -298,8 +298,10 @@ def level_params
       params[:level][param].delete_if(&:empty?) if params[:level][param].is_a? Array
     end
 
-    # Removes empty reference links which are autosaved as "" by the form
-    params[:level][:reference_links].delete_if {|link| link == ""} if params[:level][:reference_links]
+    # Reference links should be stored as an array.
+    if params[:level][:reference_links].is_a? String
+      params[:level][:reference_links] = params[:level][:reference_links].split("\r\n")
```

Comment:
What if someone added multiple new lines between reference links?

Reply:
good catch @jmkulwik ! @dmcavoy , it might make sense to use `.split(...).compact` to get rid of any nil entries that show up due to blank lines.

Reply:
we should definitely not hold this re-application PR for that, though :)